src/pages/resultsView/enrichments/AlterationEnrichmentsContainer.tsx

```diff
@@ -20,86 +18,67 @@ import AddCheckedGenes from 'pages/resultsView/enrichments/AddCheckedGenes';
 import autobind from 'autobind-decorator';
 import { EnrichmentsTableDataStore } from 'pages/resultsView/enrichments/EnrichmentsTableDataStore';
 import MiniFrequencyScatterChart from "./MiniFrequencyScatterChart";
-import EllipsisTextTooltip from "../../../shared/components/ellipsisTextTooltip/EllipsisTextTooltip";
 import FlexAlignedCheckbox from "../../../shared/components/FlexAlignedCheckbox";
+import CheckedSelect, {Option} from 'shared/components/checkedSelect/CheckedSelect';
 
 export interface IAlterationEnrichmentContainerProps {
     data: AlterationEnrichmentWithQ[];
-    totalGroup1Count: number;
-    totalGroup2Count: number;
-    group1Name?:string;
-    group2Name?:string;
-    group1Description?:string;
-    group2Description?:string;
+    groups:{name:string, description:string, displayName?:string, count:number}[]
     alteredVsUnalteredMode?:boolean;
     headerName: string;
     selectedProfile:MolecularProfile;
     store?: ResultsViewPageStore;
-    alterationType?: string;
     showCNAInTable?:boolean;
 }
 
-enum ChartType {
-    VOLCANO, FREQUENCY
-}
-
 @observer
 export default class AlterationEnrichmentContainer extends React.Component<IAlterationEnrichmentContainerProps, {}> {
 
     static defaultProps:Partial<IAlterationEnrichmentContainerProps> = {
-        group1Name: "altered group",
-        group2Name: "unaltered group",
-        alterationType: "a mutation",
         showCNAInTable: false,
         alteredVsUnalteredMode: true
     };
 
-    @computed get group1Description() {
-        return this.props.group1Description ||
-            `that have alterations in the query gene(s) that also have ${this.props.alterationType!} in the listed gene.`;
-    }
-
-    @computed get group2Description() {
-        return this.props.group2Description ||
-            `that do not have alterations in the query gene(s) that have ${this.props.alterationType!} in the listed gene.`;
-    }
-
-    @observable chartType:ChartType = ChartType.VOLCANO;
-    @observable mutualExclusivityFilter: boolean = true;
-    @observable coOccurenceFilter: boolean = true;
     @observable significanceFilter: boolean = false;
     @observable.shallow checkedGenes: string[] = [];
     @observable clickedGene: string;
     @observable.shallow selectedGenes: string[]|null;
     @observable.ref highlightedRow:AlterationEnrichmentRow|undefined;
 
-    @computed get data(): AlterationEnrichmentRow[] {
-        return getAlterationRowData(this.props.data, this.props.store ? this.props.store.hugoGeneSymbols : []);
-    }
+    @observable _enrichedGroups: string[] = this.props.groups.map(group=>group.name);
 
-    @computed get filteredData(): AlterationEnrichmentRow[] {
-        return getFilteredData(this.data, this.mutualExclusivityFilter, this.coOccurenceFilter,
-            this.significanceFilter, this.selectedGenes);
+    @computed get isTwoGroupAnalysesPossible(): boolean {
```

Comment:
I find "Possible" confusing - because it's not like you have an option to do it or not. it's more like `isTwoGroupAnalysis`, right?